src/vm/eventpipebuffermanager.cpp

```diff
@@ -57,9 +57,9 @@ EventPipeBufferManager::~EventPipeBufferManager()
                 Thread *pThread = NULL;
                 while ((pThread = ThreadStore::GetThreadList(pThread)) != NULL)
```

Comment:
@brianrob was there a reason we needed to find the Thread object that was pointing at a specific buffer list and NULL it out? It appears we only unowned a buffer when the Thread was being detached/destroyed and I would assume we could either NULL the m_pEventPipeBufferList field at that point or ignore it because it would never be read from again?


@sywhang - This loop is weird now. It is iterating over all the Thread objects, but since you don't use the Thread object in the body of the loop it is just checking the current thread's list (the thread that is executing this code, not the thread refered to by the pThread object) over and over again. Depending on what Brian says I suspect we can delete this entire while((pThread = …) loop as no longer useful, and we could set the TLS slot containing the EventPipeBufferList NULL at the same time it is unowned by a thread.